LPS-84130 9-17 ServletContextClassLoaderPool

dantewang opened the pull request.

dantewang committed:
LPS-84130 Register portal class loader to both pools

dantewang committed:
LPS-84130 Register plugin class loader with servlet context name to the new pool

dantewang committed:
LPS-84130 Apply usages of ServletContextClassLoaderPool

dantewang committed:
LPS-84130 Register servlet context name - class loader mapping in CustomServletContextHelper for osgi web bundles

dantewang committed:
LPS-84130 Deprecate ServletContext attribute PLUGIN_CLASS_LOADER since the class loader can be retrieved using ServletContextClassLoaderPool.getClassLoader

dantewang committed:
LPS-84130 Replace usages of PLUGIN_CLASS_LOADER with ServletContextClassLoaderPool

natocesarrego committed:
LPS-84622 Allow Blade form-field template support hyphen in its name.

patriciajeanperez committed:
LRQA-42696 Add another macro that unlocks a thread post

patriciajeanperez committed:
LRQA-42696 Add test that locks and unlocks a thread post

4lejandrito committed:
LPS-84329 Making portlet keys public

4lejandrito committed:
LPS-84329 Extracting notifications code to its own module

4lejandrito committed:
LPS-84329 Extracting service logic to a wrapper

dantewang committed:
LPS-84130 ClassLoader should be unregistered after undeploy events are processed

jbalsas committed:
LPS-85197 Example of how to set actions on search container rows for contextual actions. See toolbar.jsp for the possible actions for DL

4lejandrito committed:
LPS-84329 Using subscription sender to send the notification

4lejandrito committed:
LPS-84329 Enabling email notifications

4lejandrito committed:
LPS-84329 Sending body in the receiving user's locale

4lejandrito committed:
LPS-84329 Sending message in the subject and link in the body

4lejandrito committed:
LPS-84329 Removing redundant part of the message

4lejandrito committed:
LPS-84329 Adding service context

4lejandrito committed:
LPS-84329 Autogenerated

4lejandrito committed:
LPS-84329 Using real classPKs to avoid failures sending notifications

4lejandrito committed:
LPS-84329 Using the right notification types

4lejandrito committed:
LPS-84329 Adding update notification definition

4lejandrito committed:
LPS-84329 Using BaseUserNotificationTest

iiventura committed:
LPS-85239 Replace AUI._.unescape for Liferay.Util.unescape

iiventura committed:
LPS-83816 Replace _.unescape for Liferay.Util.unescape

iiventura committed:
LPS-83827 Replace _.unescape for Liferay.Util.unescape

iiventura committed:
LPS-83823 Replace _.unescape for Liferay.Util.unescape

p2kmgcl committed:
LPS-85304 Make components receive layoutData instead of layoutData.structure

p2kmgcl committed:
LPS-85304 Add createStore function to store.es.js

p2kmgcl committed:
LPS-85304 Import JSONFactoryUtil and JSONSerializer from init.jsp

p2kmgcl committed:
LPS-85304 Create store from JSP files instead of FragmentsEditor

p2kmgcl committed:
LPS-85304 Update FragmentsEditor with new Store and parameters

alee8888 committed:
LPS-84768 Remove card visualization from User portlets that do not allow images

alee8888 committed:
LPS-84768 Clean up references where display style is set to icon

alee8888 committed:
LPS-85292 Reduce top padding for the Roles Admin Web form

dantewang committed:
LPS-84130 Skip registering servlet context name - class loader mappings for bundles that have PluginContextListener

dantewang committed:
LPS-84130 Create ServletContextClassLoaderPool for servlet context name - class loader mapping

natocesarrego committed:
LPS-84622 Add unit test to cover scenario of form-field template with hyphen in its name.

iiventura committed:
LPS-83828 Replace _.unescape for Liferay.Util.unescape

p2kmgcl committed:
LPS-85304 Move store assignment to connect function

p2kmgcl committed:
LPS-85304 Move componentReady call to createStore function

iiventura committed:
LPS-83829 Created Liferay.Util.unescape function

4lejandrito committed:
LPS-84329 Getting title from the message provider

4lejandrito committed:
LPS-84329 Using util class

4lejandrito committed:
LPS-84329 SF

4lejandrito committed:
LPS-84329 Not needed

4lejandrito committed:
LPS-84329 The default date must be null otherwise entries will not be valid because the date will be in the past

4lejandrito committed:
LPS-84329 Supporting service wrappers

kiyoshilee committed:
LRQA-43558 Clean up sybase installation when directory grows too big

4lejandrito committed:
LPS-84329 Avoiding NPE

jkappler committed:
LPS-84225 Correct className and classPK

jkappler committed:
LPS-84225 Fix retrieval of fragment entry links for content pages and asset display pages

4lejandrito committed:
LPS-84329 Testing notifications also using addOrUpdate method

ealonso committed:
LPS-85304 SF

ealonso committed:
LPS-85304 Reorder

4lejandrito committed:
LPS-85197 Moving management toolbar display context to init.jsp

4lejandrito committed:
LPS-85197 Getting the available actions from the management toolbar display context

natocesarrego committed:
LPS-84622 Add little fixes.

jonathanmccann committed:
LPS-84851 Update translation

kiyoshilee committed:
LRQA-43533 Match regex when installing or uninstalling a patch

koor4356 committed:
LRQA-43519 Update database dialect message

yichenroy committed:
LRQA-41882 Add support for 7.1.x subrepo central merge

4lejandrito committed:
LPS-85197 SF

4lejandrito committed:
LPS-85197 Adding available actions according to permisssions

brianchandotcom committed:
LPS-84225 better name

michaelhashimoto committed:
LRQA-43524 Small fix for running 'setup-testable-tomcat-gc'

brianchandotcom committed:
LPS-84329 as used

brianchandotcom committed:
Record reference to liferay-portal/master-private with git-commit-portal-private as b5edb27f2a7e11e05a6931b3088ec83bf4ab7fa7

austinchiang committed:
LRQA-42696 Add test description for traceability

austinchiang committed:
LRQA-42696 SF

Commit:
LPS-84964 Fix Windows build

brianchandotcom committed:
LRQA-41882 prep next

joshchong committed:
LRQA-43551 Make text assertion of delete confirmation box optional

kyle-miho committed:
LRQA-43569 Use JSON API to make site instead of UI

kyle-miho committed:
LPS-84225 Use index to assert fragment order due to removed attribute

petershin committed:
LPS-84622 Remove line break

shuyangzhou committed:
LPS-85434 Set tomcat loggers to WARNING

dantewang committed:
LPS-84130 Fix tests

dantewang committed:
LPS-84130 Fix other possible NPEs

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

dantewang commented:
@tinatian

- For avoiding register/unregister the same class loader from both `CustomServletContextHelper` and `PluginContextListener`, see https://github.com/tinatian/liferay-portal/pull/1032/commits/bc4c3946125fb4cf862a9c6570e788f072757fd4
- For fallback, I decided to not include any in `ServletContextClassLoaderPool`. I added fixes for test failures I saw, and null protections for some usages. Other usages should either already have protections, or don't need protections.
- I modified `MainServletTestCallback` because in our old integration tests, the servlet context is mocked, and class loader is not registered to the pool. Currently there's only [one integration test](https://github.com/dantewang/liferay-portal/pull/790#issuecomment-420206655) (`BaseUpgradePortletIdTest`) failing because of this, but I think covering it in the integration test rule should be better.

tinatian commented:
@dantewang 
Could you please merge the null checking logic into the commit you applied the new pool?

So you mean with all the changes here, you still have a failing test? If yes, please fix it and resend.

Also, please run a full ci test to confirm if there are other failed tests.


Thanks.
Tina.

tinatian commented:
@dantewang 
To avoid registering twice, you did a check in ServletContextHelperRegistrationImpl which means every war will go through this path, right? If that's the case, I think it should be safe to remove the registration in PluginContextLoaderListener, we can do the registration in ServletContextHelperRegistrationImpl which means when PluginContextLoaderListener is invoked, it already has the classloader registered. This way we register earlier and unregister later than before, which means we expand the time you can see the classloader, I do not think this will affect anything, what do you think? This way the code should be more clear.

Tina.
